packages/api/src/command/medical/doc-request/append-doc-request-query-progress.ts

```diff
@@ -0,0 +1,125 @@
+import { getStatusFromProgress, Progress } from "../../../domain/medical/document-query";
+import { Patient } from "../../../domain/medical/patient";
+import { DocRequest } from "../../../domain/medical/doc-request";
+import { PatientModel } from "../../../models/medical/patient";
+import { DocRequestModel } from "../../../models/medical/doc-request";
+import { executeOnDBTx } from "../../../models/transaction-wrapper";
+import { getPatientOrFail } from "../patient/get-patient";
+import { getDocRequestOrFail } from "./get-doc-request";
+
+export type SetDocRequestQueryProgress = {
+  patient: Pick<Patient, "id" | "cxId">;
+  docRequest: Pick<DocRequest, "id">;
+  convertibleDownloadErrors?: number;
+  increaseCountConvertible?: number;
+  requestId?: string | undefined;
+} & (
+  | {
+      downloadProgress?: Progress | undefined | null;
+      convertProgress?: Progress | undefined | null;
+      reset?: false | undefined;
+    }
+  | {
+      downloadProgress: Progress;
+      convertProgress?: never;
+      reset?: true;
+    }
+);
+
+/**
+ * Appends the given properties of a doc request's document query progress.
+ * Keeps existing sibling properties when those are not provided, unless
+ * 'reset=true' is provided.
+ * @returns
+ */
+export async function appendDocRequestQueryProgress({
```

Comment:
High level: change fxn to accept patient and docRequest object, and modify the docQueryProgress within the docRequest Object. Created a new file for this fxn since changing the existing function would have broken too many things,